common/djangoapps/util/milestones_helpers.py

```diff
@@ -427,7 +428,11 @@ def add_user_milestone(user, milestone):
     """
     if not settings.FEATURES.get('MILESTONES_APP'):
         return None
-    return milestones_api.add_user_milestone(user, milestone)
+    try:
+        return milestones_api.add_user_milestone(user, milestone)
+    # to handle rare cases when milestone is already created
+    except IntegrityError:
+        pass
```

Comment:
Please add here return None, to make all return statements consistent. And also log an error message, this may be helpful for the future investigations

Reply:
updated, please check